main.cpp

```diff
@@ -117,16 +112,24 @@ int main(int argc, char *argv[])
     app.installEventFilter(eventFilter);
 
     QCommandLineParser parser;
+    QCommandLineOption logPathOption(QStringList() << "l" << "log",
+        QCoreApplication::translate("main", "Log to specified folder"),
+        QCoreApplication::translate("main", "folder"));
+    parser.addOption(logPathOption);
     parser.addHelpOption();
     parser.process(app);
 
     Monero::Utils::onStartup();
 
     // Log settings
-    Monero::Wallet::init(argv[0], "monero-wallet-gui");
-//    qInstallMessageHandler(messageHandler);
+    const QString logPath = getLogPath(parser.value(logPathOption));
+    Monero::Wallet::init(logPath.toStdString().c_str(), "");
```

Comment:
Shouldn't this be `init(argv[0], logPath.toStdString().c_str())`?

Reply:
In that case the logfile will be /path/to/bin/process_name.log and the second argument won't be used. See https://github.com/monero-project/monero/blob/master/contrib/epee/src/mlog.cpp#L68. Please correct me if I'm wrong

Reply:
Looks like the `mlog_get_default_log_path` function is implemented incorrectly. I've made a patch for it: https://github.com/monero-project/monero/pull/3736

I still think the first and second arguments to `init()` should be `argv[0]` and `logPath.toStdString().c_str()`, respectively, because that agrees with the intention expressed in `set_module_name_and_folder(const std::string& path_to_process_)`.